jenkins-plugin/src/test/java/jenkins/scm/impl/mock/SampleDVCSRepoRuleUtilsTest.java

```diff
@@ -0,0 +1,46 @@
+package jenkins.scm.impl.mock;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ */
+public class SampleDVCSRepoRuleUtilsTest {
+
+    @Test
+    public void testCopy() throws Throwable {
+        System.out.println("SampleDVCSRepoRuleUtilsTest new File('.'): " + new File(".").getAbsolutePath());
+
+        AbstractSampleDVCSRepoRule sampleDVCSRepoRule = new AbstractSampleDVCSRepoRule() {
+            @Override
+            public void init() throws Exception {
+
+            }
+        };
+
+        sampleDVCSRepoRule.before();
+
+         File mavenProjectRoot = new File("src/test/test-maven-projects/maven-jar-project");
```

Comment:
No no no no no no no, do not ever call the `File` constructor this way. Try for example

```sh
jenkinsci$ mvn -f pipeline-maven-plugin test
# oops
```

You must add test resources to `src/test/resources/` and use

```java
Paths.get(ThisClass.class.getResource("/whatever/dir").toURI())
```

(Even that is technically not guaranteed to work in general in test code, since a test runner might have test code packaged into a JAR. In the case of Surefire I think the test code will always in fact be unpacked in `target/test-classes/` so the above works because you will get a `file`-protocol URL.)